src/utils/gitHubUtils.ts

```diff
@@ -104,25 +104,22 @@ export async function getGitHubQuickPicksWithLoadMore<T>(cache: ICachedQuickPick
     }
 }
 
-export async function createGitHubRequestOptions(context: IStaticWebAppWizardContext, url: string, method: HttpMethods = 'GET'): Promise<gitHubWebResource> {
-    if (!context.accessToken) {
-        context.accessToken = await getGitHubAccessToken();
-    }
-
-    const requestOptions: gitHubWebResource = await requestUtils.getDefaultRequest(url, new TokenCredentials(context.accessToken), method);
+// a context should be passed in if many subsequential GitHub requests are made (i.e. the create SWA wizard)
+export async function createGitHubRequestOptions(context: IStaticWebAppWizardContext | undefined, url: string, method: HttpMethods = 'GET'): Promise<gitHubWebResource> {
+    const requestOptions: gitHubWebResource = await requestUtils.getDefaultRequest(url, new TokenCredentials(await getGitHubAccessToken(context)), method);
```

Comment:
I think I would prefer if you made `accessToken` a parameter. Thus it would look like one of these two options when you call it:

```
let requestOptions: gitHubWebResource = await createGitHubRequestOptions(context.accessToken, `${githubApiEndpoint}/user`);
```

```
const token: string = await getGitHubAccessToken();
const gitHubRequest: gitHubWebResource = await createGitHubRequestOptions(token, node.data.cancel_url, 'POST');
```

A few reasons I don't like the current stuff:
- It doesn't feel right to have "gitHubUtils" know about "IStaticWebAppWizardContext" (I think you agree, otherwise you wouldn't have felt the need to add a comment)
- I try to avoid cases where I have to explicitly pass `undefined` as a parameter
- It bothers me that `getGitHubAccessToken ` doesn't always get an access token

Reply:
Can't believe you're using my own comment against me...

Reply:
I do feel better about this though 😁 Thanks. I'm not sure why I didn't think of this as a solution...